pkg/ipam/funcs.go

```
@@ -15,12 +13,45 @@ type AssignIPDetails struct {
 	NetName     string
 	SubNet      *networkv1.Subnet
 	Reservelist *networkv1.ReservationList
-	FixedIP     net.IP
+	FixedIP     netip.Addr
+	// internal use only
+	excluded map[netip.Addr]bool
+	reserved map[netip.Addr]string
+}
+
+func (a *AssignIPDetails) buildExcluded() {
+	if a.excluded != nil {
+		return
+	}
+	a.excluded = make(map[netip.Addr]bool, len(a.SubNet.ExcludeAddresses))
+	for _, ipStr := range a.SubNet.ExcludeAddresses {
+		if ip, err := netip.ParseAddr(ipStr); err == nil {
+			a.excluded[ip] = true
+		}
+	}
+}
+
+func (a *AssignIPDetails) buildReserved() {
+	if a.reserved != nil {
+		return
+	}
+	a.reserved = make(map[netip.Addr]string, len(a.Reservelist.Items))
+	for _, r := range a.Reservelist.Items {
+		if res, ok := r.Spec.Reservation[a.NetName]; ok {
+			if ip, err := netip.ParseAddr(res.Address); err == nil {
```

same as above, ok to ignore err?